trie.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from piece import Piece
+import time
+
+class Trie:
+	def __init__(self):
+		self.root = TrieNode("root", None)
+
+	def get_root(self):
+		return self.root
+
+	def add_node(self, points, value):
+		p = 0
+		node = self.root
+		while points[p] in node.get_children():
+			node = node.get_children()[points[p]]
+			p += 1
+			if p == len(points):
+				node.set_value(value)
+				return node
+		while p < len(points) - 1:
+			node = node.add_child(None, points[p])
+			p += 1
+		node.add_child(value, points[p])
+		return node
+
+
+
+class TrieNode:

```

Comment:
This class looks good to me. One clarification - is `value` meant to be the boolean flag that marks that a "word" is in our "dictionary"?

Reply:
No, "value" was the name of the piece, e.g. "SQUARE", "SHORT_T". Changed to "piece" for clarity.